app/src/main/java/org/fossasia/openevent/general/MainActivity.kt

```diff
@@ -29,6 +31,9 @@ class MainActivity : AppCompatActivity() {
             navController = hostFragment.navController
         setupBottomNavigationMenu(navController)
 
+        if (!preference.getBoolean(LOCATION_SAVED, false))
+            navController.navigate(R.id.welcomeFragment)
```

Comment:
We haven't done this lately, but all logic goes in ViewModel so that it is testable.